File: wlxx-services/wlxx-system/wlxx-system-client/src/main/java/com/xzg/wlxx/system/client/enums/SexEnum.java
package com.xzg.wlxx.system.client.enums;

import com.baomidou.mybatisplus.annotation.EnumValue;
import com.fasterxml.jackson.annotation.JsonValue;

public enum SexEnum {
    MAIL(0, "男"),
    FEMALE(1, "女"),
    NULL(null, null),
    ;

    @EnumValue
    private final Integer code;
    @JsonValue
    private final String desc;

    SexEnum(Integer code, String desc) {
        this.code = code;
        this.desc = desc;
    }

}

File: wlxx-services/wlxx-system/wlxx-system-client/src/main/kotlin/com/xzg/wlxx/system/client/enums/SexEnum.kt
package com.xzg.wlxx.system.client.enums

import com.baomidou.mybatisplus.annotation.EnumValue
import com.fasterxml.jackson.annotation.JsonValue

enum class SexEnum(@field:EnumValue private val code: Int?, @field:JsonValue private val desc: String?) {
    MAIL(0, "男"),
    FEMALE(1, "女"),
    NULL(null, null)
}
